### PR TITLE
remove redundant "else" test in temperature example

### LINKED ISSUE #450: improve "else" example
**Affected document:**
https://github.com/arduino/reference-en/blob/master/Language/Structure/Control%20Structure/else.adoc

**Issue:**
IMHO "&& temperature < 70" in condition2 is unnecessary. 

**Explanation:**
condition1 is "true" if value is above "70". In this example all other "else" or "else if" conditions get ignored.
if value is below "70" first condition is "false" and second condition gets checked. This means already value is below 70 and must not be checked again.

### PR DESCRIPTION
**Pull request by Artoria2e5**
Fix #450. There is also a spelling problem with "precede".

### CODE REVIEW

File: Language/Structure/Control Structure/else.adoc [hunk 1]
```diff
@@ -57,15 +57,15 @@ Below is an extract from a code for temperature sensor system
...
 if (temperature >= 70)
 {
-  //Danger! Shut down the system
+  // Danger! Shut down the system
 }
-else if (temperature >= 60 && temperature < 70)
+else if (temperature >= 60)
 {
-  //Warning! User attention required
+  // Warning! Thorttle down, run the fans, and notify the user
```
Comment (per1234) [inline Language/Structure/Control Structure/else.adoc]: You introduced a typo here (should be spelled "Throttle"). I'm actually not convinced changing this comment provides any benefit, though I do very much agree with the change to a consistent comment style.

**Comment by robsoncouto** [comment]
I would recommend not removing the redundant part of the expression.  The code without it may be counter-intuitive for beginners. Besides, if you add more cases (temperature ranges) to the example, you probably would have to put it back.

I instead recommend just reorganizing the code as follows:

```
if (temperature < 60)
{
  // Safe! Continue usual tasks...
}
else if (temperature >= 60 && temperature < 70)
{
  // Warning! Throttle down, run the fans, and notify the user
}
else  if (temperature > 70)
{
   // Danger! Shut down the system
}
```

**Comment by Daniel-1276** [comment]
I am a beginner and raised the related issue. My point in reading the example was to understand what the difference between "if" and "else if" is. And I still think, the current example does not show the benefit of not having to use redundant checks, if the "above" statements were already checked. In the example of @robsoncouto you could also just use "if" statements IMHO.

**What about using an example something like this:**
if (temperature < 60)
{
status=good;  //means temperature is below 60
}
else if (temperature <90) 
{
status=GettingHot;  //means temperature is between 60 and 89
}
else if(temperature <100) 
{
status=DangerZone;  //means temperature is between 90 and 99
}
else
{
status=Alarm;  //means temperature is above 99
}

**Comment by robsoncouto** [comment]
Hey @per1234 , could you please fix this one as you think it is adequate?
I think that the *else* example should demonstrate well the concept of mutual exclusivity and not focus on saving a few instructions. This is just my opinion, though.

**Comment by per1234** [comment]
My concern with the current code and your suggested code even more so is not about efficiency, but ease of code maintenance. Let's imagine the programmer decides they want to reduce the danger temperature threshold to 65. With your suggested code, they need to remember to change the value in two different places. If they forget to change one of the two values (a very easy thing to do when you have a lot of code in the clause):
```c++
if (temperature < 60)
{
  // Safe! Continue usual tasks...
}
else if (temperature >= 60 && temperature < 70)
{
  // Warning! Throttle down, run the fans, and notify the user
}
else if (temperature >= 65)
{
   // Danger! Shut down the system
}
```
then the intended change in behavior was not accomplished.

Of course if you used variables instead it wouldn't be an issue, but adding variables to the example code introduces extra complexity, which is contrary to your intent of making it easier for a beginner to understand.


My suggestion is to use best practices in the code, but add comments to make it clear how the code works. Something like this:
```c++
if (temperature >= 70)
{
  // Danger! Shut down the system
}
else if (temperature >= 60)  // 60 <= temperature < 70
{
  // Warning! User attention required
}
else  // temperature < 60
{
  // Safe! Continue usual tasks...
}
```

**Comment by robsoncouto** [comment]
Updated and merged.

**Pull request merged**